### PR TITLE
Add support to add and show (create table) extra hive table properties

### LINKED ISSUE #954: Add Hive table property to for arbitrary properties
Currently only table properties explicitly listed [HiveTableProperties](https://github.com/prestosql/presto/blob/master/presto-hive/src/main/java/io/prestosql/plugin/hive/HiveTableProperties.java) are supported in Presto, but many Hive environments use extended properties for administration.  

Add a property named `extra_properties` of type `MAP(VARCHAR, VARCHAR)`.  

On read (e.g. `SHOW CREATE TABLE`) will show only the properties not mapped to existing table properties, and properties created by presto such as `presto_version` and `presto_query_id`.

On write, these properties are merged with the other properties, and if there are duplicates and error is thrown.

This is equivalent of Hive's `TBLPROPERTIES`.

### PR DESCRIPTION
**Pull request by ankitdixit**
Fixes #954

Need help in ShowQueriesRewrite, please help me converting a map to an expression

### CODE REVIEW

**Comment by ankitdixit** [comment]
https://github.com/prestosql/presto/issues/954

**Comment by ankitdixit** [comment]
Working on adding and showing properties via showCreateTable, will add merge validation post that

File: presto-main/src/main/java/io/prestosql/sql/rewrite/ShowQueriesRewrite.java [hunk 1]
```diff
@@ -411,6 +411,10 @@ private static Expression toExpression(Object value)
                         .collect(toList()));
             }
 
+            if (value instanceof Map) {
+                //TODO: How to convert map to an expression here
```
Comment (ankitdixit) [inline presto-main/src/main/java/io/prestosql/sql/rewrite/ShowQueriesRewrite.java]: There does not seem to exist a map construct which can directly be converted to expression, please suggest what to do here

Comment (findepi) [reply presto-main/src/main/java/io/prestosql/sql/rewrite/ShowQueriesRewrite.java]: `toImmutableList` 🙂

**Comment by findepi** [comment]
@ankitdixit can you please update & rebase?

this relates to https://github.com/prestosql/presto/issues/3583
cc @larandvit

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java [hunk 2]
```diff
@@ -819,6 +828,12 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
                     tableProperties.put(CSV_SEPARATOR_KEY, separator.toString());
                 });
 
+        //Extra properties
```
[the remark this answers is not in the document]
Comment (ankitdixit) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java:831]: done

**Comment by ankitdixit** [comment]
> @ankitdixit can you please update & rebase?
> 
> this relates to #3583
> cc @larandvit

I somehow missed this. updating now

**Comment by aakashnand** [comment]
@ankitdixit  Is this only for hive connector or we can add these properties to any table? I understand that currently TBLPROPERTIES is only supported in hive but can we extend this to all connectors?

**Comment by ankitdixit** [comment]
After this change, every show create table seems to have an extra line and a few tests are failing like:
`java.lang.AssertionError: expected [CREATE TABLE hive.tpch.test_comment_table (
   c1 bigint
)
WITH (
   format = 'RCBINARY'
)] but found [CREATE TABLE hive.tpch.test_comment_table (
   c1 bigint
)
WITH (
   extra_properties = map_from_entries(ARRAY[]),
   format = 'RCBINARY'
)]`
Should I change the tests or is there a way to not show extra properties line when they are empty

**Comment by findepi** [comment]
@ankitdixit we should avoid changes in the SHOW CREATE TABLE.
Did you try making `extra_properties` default to NULL instead?

**Pull request closed without merging**